snapcraft/internal/lifecycle/_runner.py

```diff
@@ -290,13 +290,20 @@ def _run_prime(self, part):
     def _reprime(self, part, hint=""):
         self._rerun_step(step=steps.PRIME, part=part, progress="Re-priming", hint=hint)
 
-    def _prepare_step(self, *, step: steps.Step, part: pluginhandler.PluginHandler):
-        common.reset_env()
+    def _handle_part_dependencies(
+        self, *, step: steps.Step, part: pluginhandler.PluginHandler
+    ) -> None:
+        # core20 uses Plugins V2 which does not require staging parts for pull
+        # like V1 Plugins do.
+        if self.project._get_build_base() == "core20" and step == steps.PULL:
+            return
```

Comment:
This will fix this for core20 only, but not for core22? Would it make sense to invert the check?

```python
        if self.project._get_build_base() not in ["core18", "core16", "core"] and step == steps.PULL:
            return
```